invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -69,9 +71,21 @@ def record_cls(self):
         """Get specified record class or from service."""
         return self._record_cls or self.get_service().record_cls
 
+    @property
+    def draft_cls(self):
+        """Get specified draft class or from service."""
+        return self._draft_cls or self.get_service().draft_cls
+
     def matches_entity(self, entity):
         """Check if the entity is a result item."""
-        return isinstance(entity, (self.item_cls, self.record_cls))
+        is_item_or_record = isinstance(entity, (self.item_cls, self.record_cls))
+        if is_item_or_record:
+            return is_item_or_record
+
+        try:
+            return isinstance(entity, self.draft_cls)
```

Comment:
after a round of a chat, I understood that this part should be probably placed in entity resolver dedicated to drafts